src/string/strlen.rs

```diff
@@ -3,12 +3,34 @@ use core::usize;
 use c_types::*;
 
 #[no_mangle]
-pub unsafe extern "C" fn strlen(s: *const c_schar) -> size_t {
-    // TODO(adam) convert to checking word-size chunks
-    for i in 0.. {
-        if *s.offset(i) == 0 {
-            return i as usize;
+pub unsafe extern "C" fn strlen(mut s: *const c_schar) -> size_t {
+    macro_rules! align { () => { core::mem::size_of::<size_t>() } }
+    macro_rules! ones { () => { !0usize } }
+    macro_rules! highs { () => { usize::wrapping_mul(ones!(), usize::MAX)/2 + 1 } }
```

Comment:
Ah yes, size_of won't be const for a while. That said, I still don't think that macros add anything here, and make it less readable. Perhaps these could be instantiated as stack variables instead?